core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -198,12 +190,19 @@ class HadoopRDD[K, V](
     if (inputFormat.isInstanceOf[Configurable]) {
       inputFormat.asInstanceOf[Configurable].setConf(jobConf)
     }
-    val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
-    val array = new Array[Partition](inputSplits.size)
-    for (i <- 0 until inputSplits.size) {
-      array(i) = new HadoopPartition(id, i, inputSplits(i))
+    // SPARK-5068:catch the exception when the path is not exists
+    try {
+      val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
+      val array = new Array[Partition](inputSplits.size)
+      for (i <- 0 until inputSplits.size) {
+        array(i) = new HadoopPartition(id, i, inputSplits(i))
+      }
+      array
+    } catch {
+      case e: InvalidInputException => {
```

Comment:
I found [Shark](https://github.com/amplab/shark/blob/master/src/main/scala/shark/execution/HadoopTableReader.scala#L152) is check for the path when the RDD was constructed(a little bit like [my last PR](https://github.com/apache/spark/pull/3891)),but i agree with @marmbrus that will slow down when constructing RDD especially a lot of partitions.